app/src/main/java/net/gsantner/opoc/util/CoolExperimentalStuff.java

```diff
@@ -141,4 +143,44 @@ public void run() {
             w.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
         }
     }
+
+    private static Thread autoScrollerThread = null;
+
+    public static void startAutoScroller(final Activity activity, final boolean isWebView) {
+        View view;
+        if (isWebView) {
+            view = activity.findViewById(R.id.document__fragment_view_webview);
+        } else {
+            view = activity.findViewById(R.id.document__fragment__edit__content_editor__scrolling_parent);
+        }
+
```

Comment:
For the logic, I dont know whats the best for average user.
Have some tries what you think fits.

So the idea is, slow & fast enough that you can finish reading the upper half screen within time.
Thats my thought on that, some things may be bigger or smaller which moves from the 50% but thats ok.

So stuff you have also is editor & viewer font size. You can include that in calculation somehow if you think it helps.